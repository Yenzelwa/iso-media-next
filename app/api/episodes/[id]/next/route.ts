import { json, errorResponse } from '@/src/lib/http';
import { validate } from '@/src/lib/validate';
import { getNextEpisodeService } from '@/src/services/episodes';
import NextEpisodeSchema from '@/schemas/episodes/NextEpisode.schema.json';

export async function GET(req: Request, { params }: { params: { id: string } }) {
  try {
    const data = await getNextEpisodeService(params.id);
    validate(NextEpisodeSchema, data);
    
    return json(data, 200);
  } catch (error) {
    if (error instanceof Error && error.message === 'Episode not found') {
      return errorResponse('not_found', 'Episode not found', {}, 404);
    }
    return errorResponse('internal', 'Internal server error', {}, 500);
  }
}