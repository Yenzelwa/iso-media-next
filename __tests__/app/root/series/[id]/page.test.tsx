import React from "react";
import { render, screen, within } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import * as nextNav from "next/navigation";
import SeriesDetail from "@/src/app/(root)/series/[id]/page";

// -------------------- Mocks --------------------

// Capture router fns for assertions
const push = jest.fn();
const replace = jest.fn();
const back = jest.fn();
const forward = jest.fn();
const refresh = jest.fn();

// Mock Next App Router hooks used by the component
jest.mock("next/navigation", () => ({
  useRouter: () => ({ push, replace, back, forward, refresh }),
  usePathname: () => "/series/2",
  useSearchParams: () => new URLSearchParams(""),
  // NOTE: your component imports useParams from react-router-dom,
  // so this useParams isn't used; we still provide it harmlessly.
  useParams: () => ({ id: "2" }),
}));

// Mock react-router-dom's useParams (component calls this)
jest.mock("react-router-dom", () => ({
  useParams: () => ({ id: "2" }),
}));

// Make lucide-react icons lightweight to keep DOM clean
jest.mock("lucide-react", () => {
  const Icon = (props: any) => <svg aria-hidden="true" {...props} />;
  return {
    Play: Icon,
    Heart: Icon,
    Share2: Icon,
    Download: Icon,
    Star: Icon,
    Clock: Icon,
    Calendar: Icon,
    ChevronLeft: Icon,
  };
});

// -------------------- Helpers --------------------

const renderPage = () => render(<SeriesDetail />);

const getSeasonButton = (container: HTMLElement, label: string) => {
  // Season buttons are the numeric buttons (1..n) in the Episodes header row
  const episodesHeader = screen.getByRole("heading", { name: /episodes/i });
  // Walk up to the container that includes the season controls
  const headerRow =
    episodesHeader.closest(".flex")?.parentElement ?? episodesHeader.parentElement!;
  const allButtons = within(headerRow).getAllByRole("button");
  const btn = allButtons.find((b) => b.textContent?.trim() === label);
  if (!btn) throw new Error(`Season button "${label}" not found`);
  return btn;
};

// -------------------- Tests --------------------

describe("SeriesDetail page", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("renders hero details: title, badges, rating, likes, year, duration, genres, description", () => {
    renderPage();

    // Title
    expect(
      screen.getByRole("heading", { name: /consciousness expansion/i })
    ).toBeInTheDocument();

    // Badges
    //expect(screen.getByRole(/series/i)).toBeInTheDocument();
    expect(screen.getByText(/3 seasons/i)).toBeInTheDocument();
    expect(screen.getByText(/24 episodes/i)).toBeInTheDocument();

    // Meta
    expect(screen.getByText("4.9")).toBeInTheDocument(); // rating
    expect(screen.getByText("267")).toBeInTheDocument(); // likes
    expect(screen.getByText("2023")).toBeInTheDocument(); // year
    expect(screen.getByText(/45-60 min per episode/i)).toBeInTheDocument(); // duration

    // Genres
    expect(screen.getByText("Spirituality")).toBeInTheDocument();
    expect(screen.getByText("Documentary")).toBeInTheDocument();
    expect(screen.getByText("Self-Development")).toBeInTheDocument();

    // Description snippet
    expect(
      screen.getByText(/transformative journey through the depths of human consciousness/i)
    ).toBeInTheDocument();
  });

  it("breadcrumb button navigates back to /series", async () => {
    const user = userEvent.setup();
    renderPage();

    const backBtn = screen.getByTestId("hero-btn");
    await user.click(backBtn);

    expect(push).toHaveBeenCalledTimes(1);
    expect(push).toHaveBeenCalledWith("/series");
  });

  it("initially lists Season 1 episodes and clicking a card navigates to that episode watch page", async () => {
    const user = userEvent.setup();
    const { container } = renderPage();

    // 6 episode cards for season 1
    const cards = container.querySelectorAll(".group.cursor-pointer");
    expect(cards.length).toBe(6);

    // Click the 3rd card (episode id 3 in mock)
    await user.click(cards[2]);
    expect(push).toHaveBeenCalledWith("/watch/3");
  });

  it("Play Series plays the first episode of the current season; switching to season 2 keeps old currentEpisode", async () => {
    const user = userEvent.setup();
    const { container } = renderPage();

    // On mount, effect sets currentEpisode to first ep of season 1 (id 1)
    const playBtn = screen.getByRole("button", { name: /play series/i });
    await user.click(playBtn);
    expect(push).toHaveBeenLastCalledWith("/watch/1");

    // Switch to Season 2 (no episodes in mock)
    const s2Btn = getSeasonButton(container, "2");
    await user.click(s2Btn);

    // Grid should be empty
    expect(container.querySelectorAll(".group.cursor-pointer").length).toBe(0);

    // Because code keeps previous currentEpisode, Play still goes to id 1
    await user.click(playBtn);
    expect(push).toHaveBeenLastCalledWith("/watch/1");

    // Switch back to Season 1; effect sets currentEpisode to first ep again
    const s1Btn = getSeasonButton(container, "1");
    await user.click(s1Btn);

    // Cards are back
    const cards = container.querySelectorAll(".group.cursor-pointer");
    expect(cards.length).toBe(6);

    // Click a different episode to update currentEpisode via handlePlayEpisode
    await user.click(cards[4]); // episode id 5
    expect(push).toHaveBeenLastCalledWith("/watch/5");

    // Now Play Series should jump to the last selected episode (id 5)
    await user.click(playBtn);
    expect(push).toHaveBeenLastCalledWith("/watch/5");
  });

  it("like button toggles active styling on and off", async () => {
    const user = userEvent.setup();
    renderPage();

    // Action bar contains Play, Like, Share, Download (in that order)
    const playBtn = screen.getByRole("button", { name: /play series/i });
    const actions = playBtn.parentElement as HTMLElement;
    const buttons = within(actions).getAllByRole("button");
    const likeBtn = buttons[1];

    // Initial (inactive)
    expect(likeBtn.className).not.toMatch(/bg-red-600/);

    // Toggle on
    await user.click(likeBtn);
    expect(likeBtn.className).toMatch(/bg-red-600/);

    // Toggle off
    await user.click(likeBtn);
    expect(likeBtn.className).not.toMatch(/bg-red-600/);
  });
});
