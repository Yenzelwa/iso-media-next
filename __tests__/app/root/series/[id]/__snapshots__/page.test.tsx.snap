// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeriesDetail page matches snapshot 1`] = `
<div
  class="bg-background text-foreground"
>
  <main
    class="pt-14 sm:pt-16 lg:pt-20 xl:pt-24"
  >
    <div
      class="relative h-[calc(100vh-3.5rem)] sm:h-[calc(100vh-4rem)] lg:h-[calc(100vh-5rem)] xl:h-[calc(100vh-6rem)] overflow-hidden"
    >
      <div
        class="absolute inset-0"
      >
        <img
          alt="Consciousness Expansion"
          class="w-full h-full object-cover scale-105 blur-sm"
          src="https://images.unsplash.com/photo-1518837695005-2083093ee35b?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=2670&q=80"
        />
        <div
          class="absolute inset-0 bg-gradient-to-r from-black/95 via-black/60 to-transparent"
        />
        <div
          class="absolute inset-0 bg-gradient-to-t from-black/90 via-transparent to-black/80"
        />
        <div
          class="absolute top-0 left-0 right-0 h-32 bg-gradient-to-b from-black/90 to-transparent"
        />
      </div>
      <div
        class="relative z-10 flex items-center h-full px-4 lg:px-16"
      >
        <div
          class="max-w-4xl"
        >
          <div
            class="flex items-center space-x-2 mb-6 text-gray-300"
          >
            <button
              class="hover:text-white transition-colors flex items-center space-x-1"
              data-testid="hero-btn"
            >
              <svg
                aria-hidden="true"
                class="w-4 h-4"
              />
              <span>
                Series
              </span>
            </button>
            <span>
              /
            </span>
            <span
              class="text-white"
            >
              Consciousness Expansion
            </span>
          </div>
          <div
            class="flex items-center space-x-3 mb-4"
          >
            <span
              class="bg-red-600 text-white text-sm px-4 py-2 rounded-lg font-bold tracking-wide"
            >
              SERIES
            </span>
            <span
              class="bg-gray-900/80 backdrop-blur-sm text-white text-sm px-3 py-2 rounded-lg"
            >
              3
               Season
              s
            </span>
            <span
              class="bg-gray-900/80 backdrop-blur-sm text-white text-sm px-3 py-2 rounded-lg"
            >
              24
               Episodes
            </span>
          </div>
          <h1
            class="text-5xl lg:text-7xl font-bold text-white mb-6 leading-tight"
          >
            Consciousness Expansion
          </h1>
          <div
            class="flex items-center space-x-6 mb-6 text-white"
          >
            <div
              class="flex items-center space-x-1"
            >
              <svg
                aria-hidden="true"
                class="w-5 h-5 text-yellow-400 fill-current"
              />
              <span
                class="font-semibold text-lg"
              >
                4.9
              </span>
            </div>
            <div
              class="flex items-center space-x-1"
            >
              <svg
                aria-hidden="true"
                class="w-5 h-5 text-red-500"
              />
              <span>
                267
              </span>
            </div>
            <div
              class="flex items-center space-x-1"
            >
              <svg
                aria-hidden="true"
                class="w-5 h-5"
              />
              <span>
                2023
              </span>
            </div>
            <div
              class="flex items-center space-x-1"
            >
              <svg
                aria-hidden="true"
                class="w-5 h-5"
              />
              <span>
                45-60 min per episode
              </span>
            </div>
          </div>
          <div
            class="flex flex-wrap gap-2 mb-8"
          >
            <span
              class="px-3 py-1 bg-gray-800/70 backdrop-blur-sm text-gray-200 rounded-full text-sm border border-gray-600/30"
            >
              Spirituality
            </span>
            <span
              class="px-3 py-1 bg-gray-800/70 backdrop-blur-sm text-gray-200 rounded-full text-sm border border-gray-600/30"
            >
              Documentary
            </span>
            <span
              class="px-3 py-1 bg-gray-800/70 backdrop-blur-sm text-gray-200 rounded-full text-sm border border-gray-600/30"
            >
              Self-Development
            </span>
          </div>
          <p
            class="text-gray-200 text-lg leading-relaxed mb-8 max-w-3xl"
          >
            Embark on a transformative journey through the depths of human consciousness. This groundbreaking series combines ancient wisdom with modern neuroscience to guide you through expanded states of awareness, meditation practices, and spiritual awakening techniques that will fundamentally shift your perspective on reality.
          </p>
          <div
            class="flex flex-wrap items-center gap-4"
          >
            <button
              class="bg-red-600 hover:bg-red-700 text-white px-8 py-4 rounded-lg font-semibold flex items-center space-x-2 transition-all duration-300 transform hover:scale-105 shadow-lg hover:shadow-red-600/25"
            >
              <svg
                aria-hidden="true"
                class="w-5 h-5"
              />
              <span>
                Play Series
              </span>
            </button>
            <button
              class="p-4 rounded-lg border-2 transition-all duration-300 border-gray-600 text-gray-300 hover:border-red-600 hover:text-red-600"
            >
              <svg
                aria-hidden="true"
                class="w-5 h-5 "
              />
            </button>
            <button
              class="p-4 rounded-lg border-2 border-gray-600 text-gray-300 hover:border-white hover:text-white transition-all duration-300"
            >
              <svg
                aria-hidden="true"
                class="w-5 h-5"
              />
            </button>
            <button
              class="p-4 rounded-lg border-2 border-gray-600 text-gray-300 hover:border-white hover:text-white transition-all duration-300"
            >
              <svg
                aria-hidden="true"
                class="w-5 h-5"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="bg-gradient-to-b from-gray-900 via-black to-gray-900 py-16"
    >
      <div
        class="px-4 lg:px-16"
      >
        <div
          class="flex items-center justify-between mb-8"
        >
          <h2
            class="text-3xl font-bold text-white flex items-center"
          >
            <div
              class="w-1 h-8 bg-red-600 rounded-sm mr-4"
            />
            Episodes
          </h2>
          <div
            class="flex items-center space-x-2"
          >
            <span
              class="text-gray-400"
            >
              Season:
            </span>
            <div
              class="flex space-x-1"
            >
              <button
                class="px-4 py-2 rounded-lg transition-all duration-300 bg-red-600 text-white"
              >
                1
              </button>
              <button
                class="px-4 py-2 rounded-lg transition-all duration-300 bg-gray-800 text-gray-300 hover:bg-gray-700"
              >
                2
              </button>
              <button
                class="px-4 py-2 rounded-lg transition-all duration-300 bg-gray-800 text-gray-300 hover:bg-gray-700"
              >
                3
              </button>
            </div>
          </div>
        </div>
        <div
          class="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-6"
        >
          <div
            class="group cursor-pointer"
            style="animation-delay: 0ms;"
          >
            <div
              class="relative overflow-hidden rounded-xl transition-all duration-500 group-hover:scale-105 shadow-xl ring-1 ring-gray-800/20 group-hover:ring-red-600/50"
            >
              <img
                alt="The Gateway to Consciousness"
                class="w-full h-48 object-cover transition-transform duration-500 group-hover:scale-110"
                src="https://images.unsplash.com/photo-1507003211169-0a1dd7228f2d?ixlib=rb-4.0.3&auto=format&fit=crop&w=1000&q=80"
              />
              <div
                class="absolute inset-0 bg-gradient-to-t from-black/90 via-black/30 to-transparent opacity-60 group-hover:opacity-80 transition-all duration-500"
              />
              <div
                class="absolute top-3 left-3"
              >
                <span
                  class="bg-red-600 text-white text-xs px-3 py-1.5 rounded-lg font-bold"
                >
                  EP 
                  1
                </span>
              </div>
              <div
                class="absolute top-3 right-3"
              >
                <span
                  class="bg-black/80 text-white text-xs px-2 py-1 rounded-lg font-medium"
                >
                  52:30
                </span>
              </div>
              <div
                class="absolute inset-0 flex items-center justify-center opacity-0 group-hover:opacity-100 transition-all duration-500"
              >
                <div
                  class="bg-red-600 text-white p-4 rounded-full shadow-2xl transform scale-75 group-hover:scale-100 transition-transform duration-300"
                >
                  <svg
                    aria-hidden="true"
                    class="w-6 h-6 fill-current"
                  />
                </div>
              </div>
              <div
                class="absolute bottom-0 left-0 right-0 p-4"
              >
                <h3
                  class="text-white font-bold text-lg mb-1"
                >
                  The Gateway to Consciousness
                </h3>
                <p
                  class="text-gray-200 text-sm line-clamp-2 mb-2"
                >
                  Introduction to expanded consciousness and the fundamental principles that govern our awareness.
                </p>
                <p
                  class="text-gray-400 text-xs"
                >
                  Released 
                  2023/06/01
                </p>
              </div>
            </div>
          </div>
          <div
            class="group cursor-pointer"
            style="animation-delay: 100ms;"
          >
            <div
              class="relative overflow-hidden rounded-xl transition-all duration-500 group-hover:scale-105 shadow-xl ring-1 ring-gray-800/20 group-hover:ring-red-600/50"
            >
              <img
                alt="Meditation and the Mind"
                class="w-full h-48 object-cover transition-transform duration-500 group-hover:scale-110"
                src="https://images.unsplash.com/photo-1544367567-0f2fcb009e0b?ixlib=rb-4.0.3&auto=format&fit=crop&w=1000&q=80"
              />
              <div
                class="absolute inset-0 bg-gradient-to-t from-black/90 via-black/30 to-transparent opacity-60 group-hover:opacity-80 transition-all duration-500"
              />
              <div
                class="absolute top-3 left-3"
              >
                <span
                  class="bg-red-600 text-white text-xs px-3 py-1.5 rounded-lg font-bold"
                >
                  EP 
                  2
                </span>
              </div>
              <div
                class="absolute top-3 right-3"
              >
                <span
                  class="bg-black/80 text-white text-xs px-2 py-1 rounded-lg font-medium"
                >
                  48:15
                </span>
              </div>
              <div
                class="absolute inset-0 flex items-center justify-center opacity-0 group-hover:opacity-100 transition-all duration-500"
              >
                <div
                  class="bg-red-600 text-white p-4 rounded-full shadow-2xl transform scale-75 group-hover:scale-100 transition-transform duration-300"
                >
                  <svg
                    aria-hidden="true"
                    class="w-6 h-6 fill-current"
                  />
                </div>
              </div>
              <div
                class="absolute bottom-0 left-0 right-0 p-4"
              >
                <h3
                  class="text-white font-bold text-lg mb-1"
                >
                  Meditation and the Mind
                </h3>
                <p
                  class="text-gray-200 text-sm line-clamp-2 mb-2"
                >
                  Explore ancient meditation techniques and their profound effects on brain structure and function.
                </p>
                <p
                  class="text-gray-400 text-xs"
                >
                  Released 
                  2023/06/08
                </p>
              </div>
            </div>
          </div>
          <div
            class="group cursor-pointer"
            style="animation-delay: 200ms;"
          >
            <div
              class="relative overflow-hidden rounded-xl transition-all duration-500 group-hover:scale-105 shadow-xl ring-1 ring-gray-800/20 group-hover:ring-red-600/50"
            >
              <img
                alt="Quantum Consciousness"
                class="w-full h-48 object-cover transition-transform duration-500 group-hover:scale-110"
                src="https://images.unsplash.com/photo-1451187580459-43490279c0fa?ixlib=rb-4.0.3&auto=format&fit=crop&w=1000&q=80"
              />
              <div
                class="absolute inset-0 bg-gradient-to-t from-black/90 via-black/30 to-transparent opacity-60 group-hover:opacity-80 transition-all duration-500"
              />
              <div
                class="absolute top-3 left-3"
              >
                <span
                  class="bg-red-600 text-white text-xs px-3 py-1.5 rounded-lg font-bold"
                >
                  EP 
                  3
                </span>
              </div>
              <div
                class="absolute top-3 right-3"
              >
                <span
                  class="bg-black/80 text-white text-xs px-2 py-1 rounded-lg font-medium"
                >
                  55:42
                </span>
              </div>
              <div
                class="absolute inset-0 flex items-center justify-center opacity-0 group-hover:opacity-100 transition-all duration-500"
              >
                <div
                  class="bg-red-600 text-white p-4 rounded-full shadow-2xl transform scale-75 group-hover:scale-100 transition-transform duration-300"
                >
                  <svg
                    aria-hidden="true"
                    class="w-6 h-6 fill-current"
                  />
                </div>
              </div>
              <div
                class="absolute bottom-0 left-0 right-0 p-4"
              >
                <h3
                  class="text-white font-bold text-lg mb-1"
                >
                  Quantum Consciousness
                </h3>
                <p
                  class="text-gray-200 text-sm line-clamp-2 mb-2"
                >
                  Dive into the intersection of quantum physics and consciousness theory.
                </p>
                <p
                  class="text-gray-400 text-xs"
                >
                  Released 
                  2023/06/15
                </p>
              </div>
            </div>
          </div>
          <div
            class="group cursor-pointer"
            style="animation-delay: 300ms;"
          >
            <div
              class="relative overflow-hidden rounded-xl transition-all duration-500 group-hover:scale-105 shadow-xl ring-1 ring-gray-800/20 group-hover:ring-red-600/50"
            >
              <img
                alt="The Power of Breathwork"
                class="w-full h-48 object-cover transition-transform duration-500 group-hover:scale-110"
                src="https://images.unsplash.com/photo-1506629905592-5b5d8a2ec57c?ixlib=rb-4.0.3&auto=format&fit=crop&w=1000&q=80"
              />
              <div
                class="absolute inset-0 bg-gradient-to-t from-black/90 via-black/30 to-transparent opacity-60 group-hover:opacity-80 transition-all duration-500"
              />
              <div
                class="absolute top-3 left-3"
              >
                <span
                  class="bg-red-600 text-white text-xs px-3 py-1.5 rounded-lg font-bold"
                >
                  EP 
                  4
                </span>
              </div>
              <div
                class="absolute top-3 right-3"
              >
                <span
                  class="bg-black/80 text-white text-xs px-2 py-1 rounded-lg font-medium"
                >
                  46:28
                </span>
              </div>
              <div
                class="absolute inset-0 flex items-center justify-center opacity-0 group-hover:opacity-100 transition-all duration-500"
              >
                <div
                  class="bg-red-600 text-white p-4 rounded-full shadow-2xl transform scale-75 group-hover:scale-100 transition-transform duration-300"
                >
                  <svg
                    aria-hidden="true"
                    class="w-6 h-6 fill-current"
                  />
                </div>
              </div>
              <div
                class="absolute bottom-0 left-0 right-0 p-4"
              >
                <h3
                  class="text-white font-bold text-lg mb-1"
                >
                  The Power of Breathwork
                </h3>
                <p
                  class="text-gray-200 text-sm line-clamp-2 mb-2"
                >
                  Learn how controlled breathing can access altered states of consciousness.
                </p>
                <p
                  class="text-gray-400 text-xs"
                >
                  Released 
                  2023/06/22
                </p>
              </div>
            </div>
          </div>
          <div
            class="group cursor-pointer"
            style="animation-delay: 400ms;"
          >
            <div
              class="relative overflow-hidden rounded-xl transition-all duration-500 group-hover:scale-105 shadow-xl ring-1 ring-gray-800/20 group-hover:ring-red-600/50"
            >
              <img
                alt="Sacred Plant Medicine"
                class="w-full h-48 object-cover transition-transform duration-500 group-hover:scale-110"
                src="https://images.unsplash.com/photo-1574482620302-1958d1aa3272?ixlib=rb-4.0.3&auto=format&fit=crop&w=1000&q=80"
              />
              <div
                class="absolute inset-0 bg-gradient-to-t from-black/90 via-black/30 to-transparent opacity-60 group-hover:opacity-80 transition-all duration-500"
              />
              <div
                class="absolute top-3 left-3"
              >
                <span
                  class="bg-red-600 text-white text-xs px-3 py-1.5 rounded-lg font-bold"
                >
                  EP 
                  5
                </span>
              </div>
              <div
                class="absolute top-3 right-3"
              >
                <span
                  class="bg-black/80 text-white text-xs px-2 py-1 rounded-lg font-medium"
                >
                  58:12
                </span>
              </div>
              <div
                class="absolute inset-0 flex items-center justify-center opacity-0 group-hover:opacity-100 transition-all duration-500"
              >
                <div
                  class="bg-red-600 text-white p-4 rounded-full shadow-2xl transform scale-75 group-hover:scale-100 transition-transform duration-300"
                >
                  <svg
                    aria-hidden="true"
                    class="w-6 h-6 fill-current"
                  />
                </div>
              </div>
              <div
                class="absolute bottom-0 left-0 right-0 p-4"
              >
                <h3
                  class="text-white font-bold text-lg mb-1"
                >
                  Sacred Plant Medicine
                </h3>
                <p
                  class="text-gray-200 text-sm line-clamp-2 mb-2"
                >
                  Explore the role of plant medicine in spiritual practices and consciousness expansion.
                </p>
                <p
                  class="text-gray-400 text-xs"
                >
                  Released 
                  2023/06/29
                </p>
              </div>
            </div>
          </div>
          <div
            class="group cursor-pointer"
            style="animation-delay: 500ms;"
          >
            <div
              class="relative overflow-hidden rounded-xl transition-all duration-500 group-hover:scale-105 shadow-xl ring-1 ring-gray-800/20 group-hover:ring-red-600/50"
            >
              <img
                alt="Energy Fields and Auras"
                class="w-full h-48 object-cover transition-transform duration-500 group-hover:scale-110"
                src="https://images.unsplash.com/photo-1419242902214-272b3f66ee7a?ixlib=rb-4.0.3&auto=format&fit=crop&w=1000&q=80"
              />
              <div
                class="absolute inset-0 bg-gradient-to-t from-black/90 via-black/30 to-transparent opacity-60 group-hover:opacity-80 transition-all duration-500"
              />
              <div
                class="absolute top-3 left-3"
              >
                <span
                  class="bg-red-600 text-white text-xs px-3 py-1.5 rounded-lg font-bold"
                >
                  EP 
                  6
                </span>
              </div>
              <div
                class="absolute top-3 right-3"
              >
                <span
                  class="bg-black/80 text-white text-xs px-2 py-1 rounded-lg font-medium"
                >
                  44:35
                </span>
              </div>
              <div
                class="absolute inset-0 flex items-center justify-center opacity-0 group-hover:opacity-100 transition-all duration-500"
              >
                <div
                  class="bg-red-600 text-white p-4 rounded-full shadow-2xl transform scale-75 group-hover:scale-100 transition-transform duration-300"
                >
                  <svg
                    aria-hidden="true"
                    class="w-6 h-6 fill-current"
                  />
                </div>
              </div>
              <div
                class="absolute bottom-0 left-0 right-0 p-4"
              >
                <h3
                  class="text-white font-bold text-lg mb-1"
                >
                  Energy Fields and Auras
                </h3>
                <p
                  class="text-gray-200 text-sm line-clamp-2 mb-2"
                >
                  Understanding the human energy field and its relationship to consciousness.
                </p>
                <p
                  class="text-gray-400 text-xs"
                >
                  Released 
                  2023/07/06
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </main>
</div>
`;
